### ISSUE TITLE
Cal/card dav cannot be accessed via dav since upgrade to OC 9.0

### ISSUE THREAD

**Issue by oliv3r** [owncloud/core#23800]
Since issue #22891 seems abandoned or this bug not as relevant, I'm filing a new bug report here.

I upgraded last weekend and similar like the original bug report keep getting 405 errors.

If the orignal issue is not exactly related to this i'll open a new one, but the description and symptoms look the same.

Using cadaver I can connect to owncloud fine, but then I get 405 errors.
% cadaver http://owncloud.fdqn.tld/remote.php/dav [0]
Authentication required for sabre/dav on server owncloud.fdqn.tld': Username: oliver Password: dav:/remote.php/dav/> cd calendars lsdav:/remote.php/dav/calendars/> ls Listing collection/remote.php/dav/calendars/': failed:
405 Method Not Allowed

Both owncloud's log and apache mention this
Exception: {"Message":"HTTP\/1.1 405 Listing members of this collection is disabled","Exception":"Sabre\DAV\Exception\MethodNotAllowed","Code":0,"Trace":"#0 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAV\/Tree.php(195): Sabre\DAVACL\AbstractPrincipalCollection->getChildren()\n#1 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(857): Sabre\DAV\Tree->getChildren('calendars')\n#2 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(911): Sabre\DAV\Server->addPathNodesRecursively(Array, Object(Sabre\DAV\PropFind))\n#3 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(336): Sabre\DAV\Server->getPropertiesForPath('calendars', Array, 1)\n#4 [internal function]: Sabre\DAV\CorePlugin->httpPropFind(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\n#5 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#6 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\Event\EventEmitter->emit('method:PROPFIND', Array)\n#7 \/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\n#8 \/var\/www\/owncloud.fqdn.tld\/htdocs\/apps\/dav\/appinfo\/v1\/caldav.php(75): Sabre\DAV\Server->exec()\n#9 \/var\/www\/owncloud.fqdn.tld\/htdocs\/remote.php(138): require_once('\/var\/www\/ownclo...')\n#10 {main}","File":"\/var\/www\/owncloud.fqdn.tld\/htdocs\/3rdparty\/sabre\/dav\/lib\/DAVACL\/AbstractPrincipalCollection.php","Line":95,"User":"oliver"}

ssl_error_log
[Thu Mar 31 08:23:55.875583 2016] [authz_core:error] [pid 8061:tid 140323518850816] [client ::1:33996] AH01630: client denied by server configuration: /var/www/owncloud.fqdn.tld/htdocs/data/.ocdata

ssl_access_log
::1 - - [31/Mar/2016:08:24:01 +0200] "PROPFIND /remote.php/dav/calendars/oliver/defaultcalendar/ HTTP/1.1" 401 355
::1 - oliver [31/Mar/2016:08:24:11 +0200] "PROPFIND /remote.php/dav/calendars/oliver/defaultcalendar/ HTTP/1.1" 401 283
.1" 401 283
::1 - oliver [31/Mar/2016:08:24:11 +0200] "PROPFIND /remote.php/dav/calendars/oliver/work/ HTTP/1.1" 401 283
::1 - oliver [31/Mar/2016:08:24:17 +0200] "PROPFIND /remote.php/dav/calendars/oliver/work/ HTTP/1.1" 401 283

data is an NFS share owned by apache:apache (the default webuser) so should be accessable just fine.
.ocdata is a 0 byte file the share: -rw-r--r-- 1 81 81 0 3 nov 17:27 .ocdata (where 81 is the other servers apache user).

Server configuration
Operating system: Gentoo
Web server: apache 2.4
Database: pgsql
PHP version: 5.6.19
ownCloud version (see ownCloud admin page): 9.0
Updated from an older ownCloud or fresh install: 8.2.2

```
Enabled:
  - activity: 2.2.1
  - calendar: true
  - comments: 0.2
  - contacts: true
  - dav: 0.1.5
  - documents: 0.13.0
  - external: 1.2
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - gallery: 14.5.0
  - libreonline: 0.1
  - mozilla_sync: 1.4
  - notes: 2.0.1
  - notifications: 0.2.3
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - firstrunwizard
  - provisioning_api
  - user_external
  - user_ldap
```

**The content of config/config.php:**
$CONFIG = array (
  'log_type' => 'owncloud',
  'loglevel' => 0,
  'log_rotate_size' => '104857600',
  'maintenance' => false,
  'instanceid' => 'xxxxx',
  'passwordsalt' => 'xxxxxx',
  'datadirectory' => '/var/www/owncloud.fqdn.tld/htdocs/data',
  'dbtype' => 'pgsql',
  'version' => '9.0.0.19',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => 'xxxxxxxxxxx',
  'installed' => true,
  'forcessl' => true,
  'ldapIgnoreNamingRules' => false,
  'appstoreenabled' => true,
  'apps_paths' =>
  array (
    0 =>
    array (
      'path' => '/var/www/owncloud.fqdn.tld/htdocs/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 =>
    array (
      'path' => '/var/www/owncloud.fqdn.tld/htdocs/localapps',
      'url' => '/localapps',
      'writable' => true,
    ),
  ),
  'theme' => '',
  'trusted_domains' =>
  array (
    0 => 'owncloud.fqdn.nl',
  ),
  'mail_smtpmode' => 'php',
  'secret' => 'xxxxxxxxx',
  'trashbin_retention_obligation' => 'auto',
  'appstore.experimental.enabled' => true,
);

My apache2 config is pretty 'standard' I would think and tries many combinations of settings, but my final version is from https://doc.owncloud.org/server/9.0/adm ... figuration

DocumentRoot "/var/www/owncloud.fqdn.tld/htdocs"
Alias /owncloud "/var/www/owncloud.fqdn.tld/htdocs/"

Header always set Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"

Options +FollowSymLinks
AllowOverride All
Require all granted

Dav off

SetEnv HOME /var/www/owncloud.fqdn.tld/htdocs"
SetEnv HTTP_HOME /var/www/owncloud.fqdn.tld/htdocs"

rewrite rules are from .htaccess and point the the correct remote.php/dav/ so that's not it either.

So I'm guessing I am missing some configuration bit somewhere or have something slightly wrong.

**Comment by DeepDiver1975** [comment]
> Using cadaver I can connect to owncloud fine, but then I get 405 errors.
> % cadaver http://owncloud.fdqn.tld/remote.php/dav [0]
> Authentication required for sabre/dav on server owncloud.fdqn.tld': Username: oliver Password: dav:/remote.php/dav/> cd calendars lsdav:/remote.php/dav/calendars/> ls Listing collection/remote.php/dav/calendars/': failed:
> 405 Method Not Allowed

this is by intention - we shall not expose all user names .... just cd into the collection with your name and you will see your calendars.

```sh
cd oliver

**Comment by RealRancor** [comment]
@oliv3r Also see https://github.com/owncloud/documentation/issues/2330 for a tracked issue to update those URLs within the documentation.

**Comment by oliv3r** [comment]
Obviously I did not have only issues with cadaver, I used it to test my connections.

I think it was a bug in davdroid, since updating davdroid to 1.x it now works again. I do not know if it was fixed by configuration changes or by changes in davdroid. I never used the webdav/dav endpoints as I use htaccess's .wellknown and my dns's caldav/carddav srv entries.

But since it now works again, I agree that the issue can be closed.
